### ISSUE TITLE
Evaluation on subsampled training set during training

### ISSUE THREAD

**Issue by zc-alexfan** [PyTorchLightning/pytorch-lightning#970]
#### What is your question?

We can evaluate on the validation set to get validation accuracy etc. Is it possible to evaluate on a subsampled version of the training set as well, for example, the training accuracy? Is there an easy wa to do it with PL?

Sorry for the simple question, but I couldn't find the answer in the documentation.

**Comment by festeh** [comment]
I'd do something like this:

```
class MyCoolModule(LightningModule):
    def __init__(self, ...):
        self._train_loader = ...
        ...

    def on_epoch_end(self):
        n_batches_to_eval = ...
        results = self.trainer.evaluate(self, [self._train_loader], n_batches_to_eval)
```

**Comment by zc-alexfan** [comment]
Thanks for the reply. I was hoping something else. For example, the current `validation_step` runs in parallel with the training. Also, I need to manually subsample the training data, instead of specifying a fraction. Maybe I should change this post to a feature request. Thanks although.
